angel_system/global_step_prediction/global_step_predictor.py

```diff
@@ -0,0 +1,596 @@
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~
+        This is a guassian filter!
+        dx - The 'geometric' distance between the 'center pixel' and the pixel
+         to sample
+    dI - The difference between the intensity of the 'center pixel' and
+         the pixel to sample
+    sigma_spatial and sigma_Intesity are constants. Higher values mean
+    that we 'tolerate more' higher value of the distances dx and dI.
+
+    Dependencies: numpy, scipy.ndimage.gaussian_filter1d
+
+    calc gaussian kernel size as: filterSize = (2 * radius) + 1; radius = floor (2 * sigma_spatial)
+    y - input data
+    """
+
+    # gaussian filter and parameters
+    radius = np.floor(2 * sSpatial)
+    filterSize = (2 * radius) + 1
+    ftrArray = np.zeros(int(filterSize))
+    ftrArray[int(radius)] = 1
+
+    # Compute the Gaussian filter part of the Bilateral filter
+    gauss = ndi.gaussian_filter1d(ftrArray, sSpatial)
+
+    # 1d data dimensions
+    width = y.size
+
+    # 1d resulting data
+    ret = np.zeros(width)
+
+    for i in range(width):
+        ## To prevent accessing values outside of the array
+        # The left part of the lookup area, clamped to the boundary
+        xmin = max(i - radius, 1)
+        # How many columns were outside the image, on the left?
+        dxmin = xmin - (i - radius)
+
+        # The right part of the lookup area, clamped to the boundary
+        xmax = min(i + radius, width)
+        # How many columns were outside the image, on the right?
+        dxmax = (i + radius) - xmax
+
+        # The actual range of the array we will look at
+        area = y[int(xmin) : int(xmax)]
+
+        # The center position
+        center = y[i]
+
+        # The left expression in the bilateral filter equation
+        # We take only the relevant parts of the matrix of the
+        # Gaussian weights - we use dxmin, dxmax, dymin, dymax to
+        # ignore the parts that are outside the image
+        expS = gauss[int((1 + dxmin)) : int((filterSize - dxmax))]
+
+        # The right expression in the bilateral filter equation
+        dy = y[int(xmin) : int(xmax)] - y[i]
+        dIsquare = dy * dy
+        expI = np.exp(-dIsquare / (sIntensity * sIntensity))
+
+        # The bilater filter (weights matrix)
+        F = expI * expS
+
+        # Normalized bilateral filter
+        Fnormalized = F / sum(F)
+
+        # Multiply the area by the filter
+        tempY = y[int(xmin) : int(xmax)] * Fnormalized
+
+        # The resulting pixel is the sum of all the pixels in
+        # the area, according to the weights of the filter
+        # ret(i,j,R) = sum (tempR(:))
+        ret[i] = sum(tempY)
+
+    return ret
+
+
+def get_gt_steps_from_gt_activities(video_dset):
+    activity_gts = video_dset.images().lookup("activity_gt")
+    step_gts = []
+    step_gts_no_background = []
+    current_step = 0
+
+    # TODO: rm these lines
+    def sanitize_str(str_: str):
+        return str_.lower().strip(" .")
```

Comment:
TODO says to remove these, I assume in lieu of importing this function from where it has been added in `angel_system.data.common.load_data`.